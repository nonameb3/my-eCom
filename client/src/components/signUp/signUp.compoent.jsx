import React, { useState } from "react";
import { compose } from "redux";
import { useDispatch } from "react-redux";
import { withRouter } from "react-router-dom";

import FormInput from "../form-input/form-input.component";
import CustomButton from "../custom-button/custom-button.component";
import { signUpStart } from "../../redux/user/user.actions";
import "./signUp.style.scss";

export function SignUpCompoent(props) {
  const dispatch = useDispatch();
  const [userCredentials, setUserCredentials] = useState({
    displayName: "",
    email: "",
    password: "",
    confirmPassword: ""
  });

  const { displayName, email, password, confirmPassword } = userCredentials;

  const handleOnSubmit = async event => {
    event.preventDefault();
    const { history } = props;
    const additionalData = userCredentials;

    if (password !== confirmPassword) {
      alert("Password and confirm'password not match!");
      return;
    }

    dispatch(signUpStart({ additionalData, history }));
  };

  const handleOnChange = event => {
    const { name, value } = event.target;
    setUserCredentials({ ...userCredentials, [name]: value });
  };

  return (
    <div className="signup">
      <h1>I do not have a account.</h1>
      <span>Sign up with your email and password</span>
      <form onSubmit={handleOnSubmit}>
        <FormInput
          type="text"
          name="displayName"
          label="Display Name"
          handleOnChange={handleOnChange}
          value={displayName}
        />
        <FormInput
          type="email"
          name="email"
          label="Email"
          handleOnChange={handleOnChange}
          value={email}
        />
        <FormInput
          type="password"
          name="password"
          label="Password"
          handleOnChange={handleOnChange}
          value={password}
        />
        <FormInput
          type="password"
          name="confirmPassword"
          label="Confirm Password"
          handleOnChange={handleOnChange}
          value={confirmPassword}
        />
        <div className="buttons">
          <CustomButton type="submit">Sign-up</CustomButton>
        </div>
      </form>
    </div>
  );
}

export default compose(withRouter)(SignUpCompoent);
