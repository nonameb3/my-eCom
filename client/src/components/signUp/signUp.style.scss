.signup{
  width: 30vw;
  display: flex;
  flex-direction: column;

  @media screen and(max-width: 800px) {
    border-top: 1.5px solid gray;
    width: unset;
  }
}