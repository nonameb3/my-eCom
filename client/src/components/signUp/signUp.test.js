import React from "react";
import { shallow } from "enzyme";

import { SignUpCompoent } from "./signUp.compoent";

let wrapper;
const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  useDispatch: () => mockDispatch
}));

wrapper = shallow(<SignUpCompoent />);
it("should render sign-up component", () => {
  expect(wrapper).toMatchSnapshot();
});
