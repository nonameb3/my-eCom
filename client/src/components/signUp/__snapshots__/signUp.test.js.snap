// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render sign-up component 1`] = `ShallowWrapper {}`;
